server/mysterium_api.go

```diff
@@ -117,21 +117,25 @@ func (mApi *mysteriumApi) SendSessionStats(sessionId string, sessionStats dto.Se
 
 func (mApi *mysteriumApi) doRequest(req *http.Request) error {
 	resp, err := mApi.http.Do(req)
-	if err == nil {
-		resp.Body.Close()
+	if err != nil {
+		log.Error(mysteriumApiLogPrefix, err)
+		return err
 	}
+	defer resp.Body.Close()
 	return parseResponseError(resp)
```

Comment:
What if parsing fails

Reply:
This doesn't parse body - it just checks status code and if it's failure, returns error.